pkg/apiserver/handlers.go

```diff
@@ -367,16 +380,24 @@ func (r *requestAttributeGetter) GetAttribs(req *http.Request) authorizer.Attrib
 
 	attribs.ReadOnly = IsReadOnlyReq(*req)
 
-	apiRequestInfo, _ := r.apiRequestInfoResolver.GetAPIRequestInfo(req)
+	// Check whether meaningful api information can be resolved for the current path
+	if r.hasApiPrefix(req) {
+		apiRequestInfo, _ := r.apiRequestInfoResolver.GetAPIRequestInfo(req)
 
-	// If a path follows the conventions of the REST object store, then
-	// we can extract the resource.  Otherwise, not.
-	attribs.Resource = apiRequestInfo.Resource
+		// If a path follows the conventions of the REST object store, then
+		// we can extract the resource.  Otherwise, not.
+		attribs.Resource = apiRequestInfo.Resource
 
-	// If the request specifies a namespace, then the namespace is filled in.
-	// Assumes there is no empty string namespace.  Unspecified results
-	// in empty (does not understand defaulting rules.)
-	attribs.Namespace = apiRequestInfo.Namespace
+		// If the request specifies a namespace, then the namespace is filled in.
+		// Assumes there is no empty string namespace.  Unspecified results
+		// in empty (does not understand defaulting rules.)
+		attribs.Namespace = apiRequestInfo.Namespace
+	}
+
+	// If a request does not fall into an api namespace/resource pattern, it's a special path.
+	if attribs.Namespace == "" && attribs.Resource == "" {
```

Comment:
I disagree with this `if` statement.  I'd rather have an `if/else` block above, so that its clear that you're either considered an `APIResourceRequest` or a `NonResourceRequest`, but never both.  I'd rather not assume that a request with the correct `APIPrefix` that fails a parse is a `NonResourcePath`.  That feels like a bad path to go down.

I could probably see an argument that says to be an `APIResourceRequest`, you must have an `APIPrefix` and an `APIGroup/APIVersion`, but I don't think keying off of `Namespace` and `Resource` is correct.